src/oracle/monitor.js

```diff
@@ -0,0 +1,429 @@
+
+    //get last offer id from offers from state
+    let lastOfferId = isNaN(state["last_offer_id"]) ? 0 : state["last_offer_id"]
+
+    //initialise variable to hold results
+    let lastResults = {
+        "last_offer_id": lastOfferId,
+        "values": state["values"] ? state["values"] : {}
+    }
+
+    //loop through offers starting from last visited index
+    for (var i = 0; i < offersBalances.offers.length; i++) {
+
+        //get current offer
+        var currentOffer = offersBalances.offers[i];
+        let id = Number(currentOffer["status"]["id"])
+
+        //if we found the last visited offer id in previous check, stop looping
+        console.log("lastOfferId", lastOfferId, "currentId", id)
+        if (id <= lastOfferId) {
+            break
+        }
+
+        //if a price invitation
+        if (currentOffer["status"]["invitationSpec"]["invitationMakerName"] == "PushPrice") {
+
+            let feed = feeds[currentOffer["status"]["invitationSpec"]["previousOffer"]]
+            let lastRound = Number(currentOffer["status"]["invitationSpec"]["invitationArgs"][0]["roundId"])
+
+            //get feeds last observed round from state
+            let lastObservedRound = state["values"].hasOwnProperty(feed) ? state["values"][feed]["round"] : 0
+
+            //if round is bigger than last observed and the offer didn't fail
+            if (lastRound > lastObservedRound && !currentOffer["status"].hasOwnProperty("error")) {
+
+                //if id is bigger than last offer id in state, set it
+                lastResults["last_offer_id"] = id
+
+                let price = Number(currentOffer["status"]["invitationSpec"]["invitationArgs"][0]["unitPrice"]) / amountsIn[feed]
+
+                //fill results variable
+                lastResults["values"][feed] = {
+                    price: price,
+                    id: id,
+                    round: lastRound
+                }
+                state = lastResults
+
+                //get latest feed price
+                let feedPrice = await queryPrice(feed)
+                //update metrics
+                updateMetrics(oracleDetails["oracleName"], oracle, feed, price, id, feedPrice, lastRound)
+
+            }
+        }
+    }
+
+    //loop through balances
+    for (var i = 0; i < offersBalances.balances.length; i++) {
+        let currentBalance = offersBalances.balances[i]
+
+        var brand = currentBalance.brand.iface.split(" ")[1]
+        if (brand.includes("BLD") || brand.includes("IST")) {
+            var value = Number(currentBalance.value)
+            updateBalanceMetrics(oracleDetails["oracleName"], oracle, brand, value)
+        }
+    }
+
+    return lastResults["last_offer_id"] != lastOfferId ? lastResults : state
+}
+
+/**
+ * Function to read the latest monitoring state from file
+ * @returns latest monitoring state
+ */
+const readMonitoringState = () => {
+    //try to read from file
+    try {
+        return readJSONFile(STATE_FILE)
```

Comment:
This needs to be changed to be from DB now right?

Reply:
We will keep a state file for the monitoring script and not use a DB because the state in the monitoring script is simple and there is no need for a DB